### PR TITLE
Do not prepend a "\" to the type, if it is already Fully Qualified

### LINKED ISSUE #7023: Incorrect behavior of Rector\PHPStanStaticTypeMapper\TypeMapper\ObjectTypeMapper
# Bug Report

| Subject        | Details              |
| :------------- | :--------------------|
| Rector version | last dev-master      |
| Installed as   | composer dependency  |

## Minimal PHP Code Causing Issue

See https://getrector.org/demo/2f6d538b-5797-463b-8430-6e2fa32253d3

```php
<?php

namespace Acme\Entity\Image;

use Doctrine\ORM\Mapping as ORM;
use Acme\Entity\User;

/**
 * @ORM\Entity
 */
class UserImage extends Image
{

    /**
     * @ORM\ManyToOne(targetEntity="\Acme\Entity\User", inversedBy="images")
     * @ORM\JoinColumn(name="data_id", nullable=false)
     * @var User
     */
    private $user;

    public function __construct(User $user)
    {
        parent::__construct();
        $this->user = $user;
    }

    public function getUser(): User
    {
        return $this->user;
    }
}
```

### Responsible rules

* `TypedPropertyFromToOneRelationTypeRector`

## Expected Behavior

Minimum: No double \\ 
Optimal: Re-User the existing use-Import for the User-Class in the changed @var-Node

### PR DESCRIPTION
**Pull request by mkrauser**
If the type is already fully qualified, we do not need to prepend a "\"

Fixes https://github.com/rectorphp/rector/issues/7023

### CODE REVIEW

**Comment by samsonasik** [comment]
could you add test for it? It should be at rules-tests/Doctrine/Rector/Property/TypedPropertyFromToOneRelationTypeRector/Fixture

**Comment by samsonasik** [comment]
Could you add test to https://github.com/rectorphp/rector-doctrine/tree/main/tests/Rector/Property/TypedPropertyFromToOneRelationTypeRector/Fixture ?

**Comment by samsonasik** [comment]
Or add test here in rector-src, under tests/Issues with use the service rule definition from rector-doctrine service https://github.com/rectorphp/rector-src/tree/main/tests/Issues

**Comment by mkrauser** [comment]
> Or add test here in rector-src, under tests/Issues with use the service rule definition from rector-doctrine service https://github.com/rectorphp/rector-src/tree/main/tests/Issues

Sure, will do.
I was already looking for a TestCase for this class, but I could not find one.

**Comment by mkrauser** [comment]
I've implemented a Test for this. But the test is only valid for PHP-Versions >= 7.4.
For PHP <= 7.3 theres basically the same issue, but with a different fix.
Is there some way to add a test specific for PHP 7.3?

### Important:
The underlaying issue for PHP > 7.4 is, that to create a FullyQualified-Object the parameter must not have a leading Backslash. I suspect, that this issue will be there more often. 

see https://github.com/nikic/PHP-Parser/issues/523 for details.

**Comment by samsonasik** [comment]
You can configure php version feature in test config, eg: `PhpVersionFeature::TYPED_PROPERTIES - 1` which means < 7.4

```php
<?php

declare(strict_types=1);

use Rector\Core\Configuration\Option;
use Rector\Core\ValueObject\PhpVersionFeature;
use Rector\Doctrine\Rector\Property\TypedPropertyFromToOneRelationTypeRector;
use Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator;

return static function (ContainerConfigurator $containerConfigurator): void {
    $parameters = $containerConfigurator->parameters();
    $parameters->set(Option::PHP_VERSION_FEATURES, PhpVersionFeature::TYPED_PROPERTIES - 1);

    $services = $containerConfigurator->services();
    $services->set(TypedPropertyFromToOneRelationTypeRector::class);
};
```

**Comment by mkrauser** [comment]
Test for PHP 7.3 added.

Since this PR kind of fixes two separate issues, should I split this PR also?

**Comment by samsonasik** [comment]
I think it is fine to have it in single PR since it is related.

File: packages/PHPStanStaticTypeMapper/TypeMapper/ObjectTypeMapper.php [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 namespace Rector\PHPStanStaticTypeMapper\TypeMapper;
 
+use PHP_CodeSniffer\Reports\Full;
```
Comment (samsonasik) [inline packages/PHPStanStaticTypeMapper/TypeMapper/ObjectTypeMapper.php]: unused use can be removed.

Comment (mkrauser) [reply packages/PHPStanStaticTypeMapper/TypeMapper/ObjectTypeMapper.php]: Thank you! Just removed it

**Review by samsonasik — APPROVED** [review]

**Comment by samsonasik** [comment]
Thank you @mkrauser

**Pull request merged**